include/ry/codegen.hpp

```
@@ -55,12 +55,26 @@ class CodeGen {
         ResultPtrWithListMeta // ResultPtr + type_meta_[TM_ListElem] annotation
     };
 
+    struct NativeDispatchEntry;  // forward declaration for member-fn pointer
+
+    // Member function pointer for custom emitter escape hatch.
+    using CustomEmitterFn = llvm::Value *(CodeGen::*)(const CallExpr &);
+
+    // Post-call type metadata annotation for TM_ListElem.
+    enum class ListElemMeta : uint8_t { None, I8, Ptr };
+
     struct NativeDispatchEntry {
```

Clang (14.0.6)

[warning] constructor does not initialize these fields: fn_name, rt_suffix, wrapping, arity, out_param_type

(cppcoreguidelines-pro-type-member-init)